### PR TITLE
Added support for lerpDuration()

### PR DESCRIPTION
**Pull request by matthew-carroll**
Adds the ability to linearly interpolate Durations.

### CODE REVIEW

**Comment by matthew-carroll** [comment]
@cbracken I moved the `lerpDuration` implementation to Foundation and I used your `lerpDouble` tests for inspiration.

@Hixie I placed `lerpDuration` within `basic_types.dart`. Does that seem like a reasonable place to put it? And if so, does it also seem reasonable to place the tests in a new `basic_types_test.dart`?

WRT to the tests, are there specific edge-case values that you believe are important here? I know you mentioned large numbers, infinity, and NaN WRT to `lerpDouble`. I altered this implementation to only accept non-null `Duration`s because I think that's a reasonable restriction, which then defers the handling of various numerical values to `Duration`, itself. As for the `t` value, I'm not sure what the expected behavior should be for infinity or NaN. I suppose infinity should produce infinity, and NaN sounds like it deserves an exception. Thoughts?

**Comment by Hixie** [comment]
This all looks great. Durations can't be NaN or Infinity so no worries there. Supporting non-null only seems fine, it's not clear what a null Duration would mean. basic_types.dart seems fine.

**Review by Hixie — APPROVED** [review]
LGTM, but travis is angry.

...I really need to update this stamp to refer to tests in general since we don't use Travis anymore...

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/foundation/basic_types.dart [hunk 1]
```diff
@@ -238,3 +238,10 @@ class Factory<T> {
     return 'Factory(type: $type)';
   }
 }
+
+/// Linearly interpolate between two `Duration`s.
+Duration lerpDuration(Duration a, Duration b, double t) {
+  return Duration(
+    microseconds: (a.inMicroseconds + (b.inMicroseconds - a.inMicroseconds) * t).round(),
+  );
+}
```
Comment (cbracken) [inline packages/flutter/lib/src/foundation/basic_types.dart:247]: Here and below: Very minor nit, source files should [end with a linefeed](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline). That said, that's probably more pedantic than anything -- I'm 99% certain that the entire Dart toolchain consistently handles this just fine without it. But not a bad idea anyway.

**Review by cbracken — APPROVED** [review]
LGTM, but looks like presubmit is failing.

Animated GIF of a delivery truck approaching a highway overpass, then failing to meet the height clearance requirement and crashing into it

**Pull request merged**